### PR TITLE
use useMount instead of sync once

### PR DESCRIPTION
**Pull request by istarkov**
## Description

useSyncInitializeOnce is incompatibel with modern React especially in dev mode.
useUnmount can't be used with it and any hooks with cleanup would give the same issue.

Should affect only devmode

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/canvas/canvas.tsx [hunk 1]
```diff
@@ -167,7 +167,7 @@ export const Canvas = ({ params }: CanvasProps): JSX.Element | null => {
   const components = new Map(
     Object.entries({ ...baseComponents, ...remixComponents })
   ) as Components;
-  useSyncInitializeOnce(() => {
+  useMount(() => {
```
Comment (kof) [inline apps/builder/app/canvas/canvas.tsx:170]: this is basically useEffect(effect, []);

Comment (istarkov) [reply apps/builder/app/canvas/canvas.tsx:170]: useMount useUnmount is std hooks in react-use 
they are more clear than `useEffect(() => {}, [])`
They do not interfere with eslint i.e. add dependency x into `[]`

**Comment by kof** [comment]
I don't really understand how this is solving it in a different way

**Comment by kof** [comment]
How was the previous version failing exactly?

**Review by kof — APPROVED** [review]

**Comment by istarkov** [comment]
> I don't really understand how this is solving it in a different way

useSyncInitializeOnce  should not be used at all, as its not working with any react cleanup usually placed in useEffect return callback.

Because now in dev mode, every useEffect is called twice

**Comment by istarkov** [comment]
Current issue that I added cleanup and killed dev mode.

**Review by TrySound — APPROVED** [review]

**Pull request merged**